mongo/bulk_write.go

```diff
@@ -298,6 +297,9 @@ func createDeleteDoc(
 ) (bsoncore.Document, error) {
 	f, err := marshal(filter, bsonOpts, registry)
 	if err != nil {
+		if filter == nil {
+			return nil, fmt.Errorf("%w: filter is required", err)
+		}
```

Comment:
update the error message when the filter is not given.

Reply:
Is there a reason to nest this nil filter check in this error handling block? Can we move this check to the beginning of `marshal`?

Optional: Consider a clearer error message, like
> delete filter cannot be nil